pkg/adapter/containers.go

```diff
@@ -694,3 +694,45 @@ func (r *LocalRuntime) UnpauseContainers(ctx context.Context, cli *cliconfig.Unp
 	}
 	return pool.Run()
 }
+
+// Cleanup any leftovers bits of stopped containers
+func (r *LocalRuntime) CleanupContainers(ctx context.Context, cli *cliconfig.CleanupValues) ([]string, map[string]error, error) {
+	var (
+		ok       = []string{}
+		failures = map[string]error{}
+	)
+
+	ctrs, err := shortcuts.GetContainersByContext(cli.All, cli.Latest, cli.InputArgs, r.Runtime)
+	if err != nil {
+		return ok, failures, err
+	}
+
+	for _, ctr := range ctrs {
+		if cli.Remove {
+			err = removeContainer(ctx, ctr, r)
+		} else {
+			err = cleanContainer(ctx, ctr, r)
+		}
+
+		if err == nil {
+			ok = append(ok, ctr.ID())
+		} else {
+			failures[ctr.ID()] = err
+		}
+	}
+	return ok, failures, nil
+}
+
+func removeContainer(ctx context.Context, ctr *libpod.Container, runtime *LocalRuntime) error {
+	if err := runtime.RemoveContainer(ctx, ctr, false, true); err != nil {
+		return errors.Wrapf(err, "failed to cleanup and remove container %v", ctr.ID())
+	}
+	return nil
+}
+
+func cleanContainer(ctx context.Context, ctr *libpod.Container, runtime *LocalRuntime) error {
```

Comment:
I'd prefer `cleanupContainer` to make the relationship to the API call more explicit

Reply:
changed, no strong feelings here